core/server/utils/pre-check.js

```diff
@@ -0,0 +1,47 @@
+var validVersions = process.env.npm_package_engines_node.split(' || '),
```

Comment:
**this is a general suggestion**
put the script into `core/server/utils/npm/preinstall.js`
just having the `pre-check.js` inside `utils` folder looks like it's a server util.
And if we name the script `preinstall.js` it fits the npm wording.